graphbolt/src/csc_sampling_graph.cc

```diff
@@ -115,5 +117,35 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::InSubgraph(
           : torch::nullopt);
 }
 
+c10::intrusive_ptr<CSCSamplingGraph>
+CSCSamplingGraph::BuildGraphFromSharedMemoryTensors(
+    std::tuple<
+        SharedMemoryPtr, SharedMemoryPtr,
+        std::vector<torch::optional<torch::Tensor>>>&& shared_memory_tensors) {
```

Comment:
why `std::vector<>` is used instead of `std::tuple<>` or explicitly defined struct? access via index like `0`, `1` looks weird to me. Is this the best practice? just curious.

Reply:
`vector` is used here considering we may add more tensors in CSCSamplingGraph while the tensors loaded from shared memory are also in a vector.

Reply:
but if we add more tensors, we have to manually update in line 127 with `optional_tensors[4/5/6]`? Is it possible to remove such explicit index access? a new constructor or variadic template(parameters pack)?

Reply:
I feel the index access of a vector is unavoidable because we always need to load the tensors of a class from a vector. What do you mean by a new constructor or variadic template?

Reply:
i don't know yet. could the size of vector be determined in compile stage?